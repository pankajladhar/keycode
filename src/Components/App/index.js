import React, { Component } from 'react';

class App extends Component {
  constructor() {
    super();
    this.state = {
      keyCode: "Press key",
      key: "---",
      which:"---",
      code: "---"
    };
  }

  componentDidMount() {
    document.addEventListener('keydown', (e) => {
      this.setState({
        keyCode: e.keyCode,
        key: e.key,
        which:e.which,
        code: e.code
      })
    })
  }

  componentWillUnmount() {
    document.removeEventListener('keydown')
  }

  render() {
    const { keyCode, key,which, code} = this.state
    return (
      <div className="App">
        <span className="App__KeyCode">{this.state.keyCode}</span>
        <div className="Box__Container">
          <div className="Box">
            <h2>e.key</h2>
            <span>{key}</span>
          </div>
          <div className="Box">
            <h2>e.which</h2>
            <span>{which}</span>
          </div>
          <div className="Box">
            <h2>e.code</h2>
            <span>{code}</span>
          </div>
        </div>
      </div>
    );
  }
}

export default App