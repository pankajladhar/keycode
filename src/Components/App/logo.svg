<?xml version="1.0" encoding="UTF-8"?>
<svg width="78px" height="57px" viewBox="0 0 78 57" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 49 (51002) - http://www.bohemiancoding.com/sketch -->
    <title>Group</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Group" transform="translate(-1.000000, -10.000000)">
            <path d="M1.7578125,41.9882812 L1.7578125,34.9570312 C3.18360088,34.8789059 4.25780889,34.6640643 4.98046875,34.3125 C5.70312861,33.9609357 6.32812236,33.3701213 6.85546875,32.5400391 C7.38281514,31.7099568 7.74413965,30.6699281 7.93945312,29.4199219 C8.09570391,28.4824172 8.17382812,26.8515741 8.17382812,24.5273438 C8.17382812,20.7382623 8.34960762,18.0966872 8.70117188,16.6025391 C9.05273613,15.108391 9.68749541,13.9072311 10.6054687,12.9990234 C11.5234421,12.0908158 12.8613193,11.3730495 14.6191406,10.8457031 C15.8105528,10.4941389 17.6855341,10.3183594 20.2441406,10.3183594 L21.796875,10.3183594 L21.796875,17.3203125 C19.6288954,17.3203125 18.2373078,17.4423816 17.6220703,17.6865234 C17.0068329,17.9306653 16.5527358,18.3017553 16.2597656,18.7998047 C15.9667954,19.2978541 15.8203125,20.1523377 15.8203125,21.3632812 C15.8203125,22.5937562 15.7324228,24.9277172 15.5566406,28.3652344 C15.4589839,30.2988378 15.2050802,31.8662049 14.7949219,33.0673828 C14.3847636,34.2685607 13.8623079,35.2597617 13.2275391,36.0410156 C12.5927703,36.8222695 11.6211003,37.6328083 10.3125,38.4726562 C11.4648495,39.1367221 12.4072229,39.9130815 13.1396484,40.8017578 C13.872074,41.6904341 14.4335918,42.7695249 14.8242187,44.0390625 C15.2148457,45.3086001 15.4687494,47.0078019 15.5859375,49.1367187 C15.7226569,52.3789225 15.7910156,54.4492143 15.7910156,55.3476562 C15.7910156,56.6367252 15.9472641,57.5351537 16.2597656,58.0429687 C16.5722672,58.5507838 17.0458953,58.9365221 17.6806641,59.2001953 C18.3154329,59.4638685 19.6874895,59.5957031 21.796875,59.5957031 L21.796875,66.6269531 L20.2441406,66.6269531 C17.6074087,66.6269531 15.5859445,66.4169943 14.1796875,65.9970703 C12.7734305,65.5771463 11.5869189,64.8789111 10.6201172,63.9023437 C9.65331548,62.9257764 9.00390791,61.7197337 8.671875,60.2841797 C8.33984209,58.8486256 8.17382812,56.5879061 8.17382812,53.5019531 C8.17382812,49.9081852 8.01757969,47.5742241 7.70507812,46.5 C7.27538848,44.9374922 6.62598091,43.8193393 5.75683594,43.1455078 C4.88769097,42.4716763 3.55469648,42.085938 1.7578125,41.9882812 Z M78.046875,41.9882812 C76.6210866,42.0664066 75.5468786,42.2812482 74.8242188,42.6328125 C74.1015589,42.9843768 73.4814479,43.5751912 72.9638672,44.4052734 C72.4462865,45.2353557 72.0800792,46.2753844 71.8652344,47.5253906 C71.7089836,48.4628953 71.6308594,50.0839729 71.6308594,52.3886719 C71.6308594,56.1777533 71.4599626,58.8242112 71.1181641,60.328125 C70.7763655,61.8320388 70.146489,63.0380814 69.2285156,63.9462891 C68.3105423,64.8544967 66.9628995,65.572263 65.1855469,66.0996094 C63.9941347,66.4511736 62.1191534,66.6269531 59.5605469,66.6269531 L58.0078125,66.6269531 L58.0078125,59.5957031 C60.0976667,59.5957031 61.4648405,59.4638685 62.109375,59.2001953 C62.7539095,58.9365221 63.2275375,58.5556666 63.5302734,58.0576172 C63.8330093,57.5595678 63.9892578,56.7148497 63.9990234,55.5234375 C64.0087891,54.3320253 64.0917961,52.0566574 64.2480469,48.6972656 C64.3457036,46.6660055 64.6093729,45.0400452 65.0390625,43.8193359 C65.4687521,42.5986267 66.0400355,41.5634808 66.7529297,40.7138672 C67.4658239,39.8642536 68.3789007,39.1171907 69.4921875,38.4726562 C68.0468678,37.5351516 66.9921908,36.617192 66.328125,35.71875 C65.4101517,34.4296811 64.7851579,32.7890725 64.453125,30.796875 C64.2382802,29.4296807 64.0820317,26.4218982 63.984375,21.7734375 C63.9648437,20.3085864 63.833009,19.3320337 63.5888672,18.84375 C63.3447253,18.3554663 62.9052766,17.9794935 62.2705078,17.7158203 C61.635739,17.4521471 60.2148548,17.3203125 58.0078125,17.3203125 L58.0078125,10.3183594 L59.5605469,10.3183594 C62.1972788,10.3183594 64.218743,10.5234354 65.625,10.9335938 C67.031257,11.3437521 68.2128858,12.0419873 69.1699219,13.0283203 C70.1269579,14.0146534 70.7763655,15.2255788 71.1181641,16.6611328 C71.4599626,18.0966869 71.6308594,20.3574064 71.6308594,23.4433594 C71.6308594,27.017596 71.7773423,29.351557 72.0703125,30.4453125 C72.5000021,32.0078203 73.1542925,33.1259732 74.0332031,33.7998047 C74.9121138,34.4736362 76.249991,34.8593745 78.046875,34.9570312 L78.046875,41.9882812 Z" id="{-}" fill="#ff793f"></path>
            <g id="f7e7f6683b935a693972f5f8f4c8db59" transform="translate(40.000000, 40.500000) rotate(-45.000000) translate(-40.000000, -40.500000) translate(20.000000, 21.000000)" fill="#F9BB13" fill-rule="nonzero">
                <path d="M38.7510204,0 L33.7510204,0 C33.4489796,0 33.155102,0.10744898 32.9265306,0.30244898 L15.6693878,15.0508163 C14.7755102,14.8677551 13.3959184,14.625 12.4979592,14.625 C5.60816327,14.625 0,20.0929592 0,26.8144898 C0,33.5360204 5.60816327,39 12.5020408,39 C19.3959184,39 25.0040816,33.5320408 25.0040816,26.8105102 C25.0040816,24.8127551 24.6122449,23.4756122 24.1714286,22.3653061 L28.4244898,20.9565306 C28.9428571,20.7854082 29.2857143,20.3038776 29.2734694,19.7745918 L29.1673469,15.1463265 L33.9428571,14.8239796 C34.5836735,14.7802041 35.0857143,14.2708163 35.1061224,13.6460204 L35.2816327,8.0905102 L39.0204082,7.28265306 C39.5918367,7.15928571 40,6.66183673 40,6.0927551 L40,1.2177551 C40,0.545204082 39.4408163,0 38.7510204,0 Z M8.75102041,26.8144898 C10.2530612,26.8144898 11.5469388,27.6740816 12.1469388,28.9157143 C12.3755102,29.3892857 12.5020408,29.9145918 12.5020408,30.4717347 C12.5020408,32.4893878 10.8204082,34.1289796 8.75102041,34.1289796 C6.68163265,34.1289796 5,32.4893878 5,30.4717347 C5,28.4540816 6.68163265,26.8144898 8.75102041,26.8144898 Z" id="Shape"></path>
            </g>
        </g>
    </g>
</svg>