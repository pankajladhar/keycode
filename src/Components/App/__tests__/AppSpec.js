import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './../index';

describe('App', () => {
  it('renders correctly when no key is pressed', () => {
    let tree = renderer.create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('componentDidMount should be called', () => {
    const methodNameFake = jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);
    expect(methodNameFake).toHaveBeenCalledTimes(1);
  })

  it('setKeyCode should set state properly', () => {
    let component = renderer.create(<App />);
    const instance = component.getInstance()
    instance.setKeyCode({
      keyCode: "keyCode",
      key: "key",
      which: "which",
      code: "code"
    })

    expect(instance.state.keyCode).toEqual("keyCode")
    expect(instance.state.key).toEqual("key")
    expect(instance.state.which).toEqual("which")
    expect(instance.state.code).toEqual("code")
  })

  it('renders correctly when key is pressed', () => {
    let component = renderer.create(<App />);
    const instance = component.getInstance()
    instance.setKeyCode({
      keyCode: "keyCode",
      key: "key",
      which: "which",
      code: "code"
    })

    expect(component.toJSON()).toMatchSnapshot();
  });
})

