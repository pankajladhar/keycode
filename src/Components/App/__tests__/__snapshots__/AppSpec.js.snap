// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly when key is pressed 1`] = `
<div
  className="App"
>
  <span
    className="App__KeyCode"
  >
    keyCode
  </span>
  <div
    className="Box__Container"
  >
    <div
      className="Box"
    >
      <h2>
        e.key
      </h2>
      <span>
        key
      </span>
    </div>
    <div
      className="Box"
    >
      <h2>
        e.which
      </h2>
      <span>
        which
      </span>
    </div>
    <div
      className="Box"
    >
      <h2>
        e.code
      </h2>
      <span>
        code
      </span>
    </div>
  </div>
</div>
`;

exports[`App renders correctly when no key is pressed 1`] = `
<div
  className="App"
>
  <span
    className="App__KeyCode"
  >
    Press key
  </span>
</div>
`;
